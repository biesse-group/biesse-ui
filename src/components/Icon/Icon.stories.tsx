import { type Meta, type StoryObj } from "@storybook/react";

import { Icon, type IconProps } from "./Icon";

const meta: Meta<typeof Icon> = {
  title: "Icons/Icon",
  component: Icon,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof Icon>;

const defaultArgs: IconProps = {
  name: "careers",
};

export const Large: Story = {
  args: {
    ...defaultArgs,
    size: "lg",
  },
};

export const Medium: Story = {
  args: {
    ...defaultArgs,
    size: "md",
  },
};

export const Small: Story = {
  args: {
    ...defaultArgs,
    size: "sm",
  },
};

export const ExtraSmall: Story = {
  args: {
    ...defaultArgs,
    size: "xs",
  },
};
