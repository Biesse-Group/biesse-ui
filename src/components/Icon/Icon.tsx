import { FC } from "react";
import styled from "styled-components";

import iconsMap from "./icons-map";

export type IconName = keyof typeof iconsMap;

export interface IconProps {
  /**
   * Optional component class name
   */
  className?: string;
  /**
   * Id of the icon you want to use
   */
  name: IconName;
  /**
   * Which from the default sizes of the icon you want to use
   */
  size?: "xs" | "sm" | "md" | "lg" | string;
  /**
   * Color fo the icon.
   * if not specified becomes inherit.
   */
  color?: string;
  testId?: string;
  className?: string;
}

const getIconSize = (size?: IconProps["size"]) => {
  switch (size) {
    case "xs":
      return "20px";
    case "sm":
      return "30px";
    case "lg":
      return "80px";
    case "md":
      return "50px";
    default:
      return size;
  }
};

const IconRoot = styled.div<Omit<IconProps, "name">>`
  color: ${(props) => props.color || "inherit"};

  > svg {
    height: ${(props) => getIconSize(props.size)};
    color: ${(props) => props.color || "inherit"};
  }
`;

export const Icon: FC<IconProps> = ({ name, size, color, className, testId, ...props }) => {
  const IconComponent = iconsMap[name];
  return (
    <IconRoot size={size} color={color} data-testid={testId} className={className}>
      <IconComponent {...props} />
    </IconRoot>
  );
};
