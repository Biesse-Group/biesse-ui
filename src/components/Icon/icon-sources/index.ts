export * from "./ArrowLeft";
export * from "./ArrowRight";
export * from "./Call";
export * from "./Careers";
export * from "./ChevronRight";
export * from "./Close";
export * from "./Contacts";
export * from "./Country";
export * from "./Customer";
export * from "./Download";
export * from "./Email";
export * from "./Hamburger";
export * from "./Innovation";
export * from "./Location";
export * from "./MaterialComposite";
export * from "./MaterialGlass";
export * from "./MaterialMetal";
export * from "./MaterialStone";
export * from "./MaterialWood";
export * from "./Picture";
export * from "./Play";
export * from "./FastRepair";
export * from "./RoundAngle";
export * from "./Search";
export * from "./SpareParts";
export * from "./Sustainability";
export * from "./VideoPill";
export * from "./Linkedin";
