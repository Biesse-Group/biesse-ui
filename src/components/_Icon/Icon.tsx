import { FC } from "react";
import styled, { css, useTheme } from "styled-components";

import iconsMap from "./icons-map";

export interface IconProps {
  /**
   * Id of the icon you want to use
   */
  name: keyof typeof iconsMap;
  /**
   * Which from the default sizes of the icon you want to use
   */
  size?: "small" | "medium" | "large";
  /**
   * Custom size
   */
  height?: string;
  /**
   * Color of icon
   */
  color?: string;
}

const getIconSize = (size?: string) => {
  switch (size) {
    case "small":
      return css`
        height: 30px;
      `;
    case "large":
      return css`
        height: 80px;
      `;
    default:
      return css`
        height: 50px;
      `;
  }
};

const IconRoot = styled.div<Pick<IconProps, "size" | "height">>`
  > svg {
    height: 100%;
  }

  ${(props) =>
    props.height
      ? css`
          height: ${props.height};
        `
      : getIconSize(props.size)}
`;

export const Icon: FC<IconProps> = ({ name, size, height, ...props }) => {
  const IconComponent = iconsMap[name];
  const theme = useTheme();
  props.color = props.color || theme.color.primary;
  return (
    <IconRoot size={size} height={height}>
      <IconComponent {...props} />
    </IconRoot>
  );
};
