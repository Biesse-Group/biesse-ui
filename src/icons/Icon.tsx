import React, { FC } from "react";
import styled from "styled-components";

import getIconSize from "./get-icon-size";
import { IconProps } from "./IconProps";
import * as IconSources from "./icons-sources";

const IconRoot = styled.div<Omit<IconProps, "name">>`
  & svg {
    width: 100%;
  }

  ${(props) => getIconSize(props.size)}
`;

const renderIconImg = (name: string) => {
  switch (name) {
    case "search":
      return <IconSources.SearchIcon />;
    default:
      return <IconSources.PlaceHolderIcon />;
  }
};

export const Icon: FC<IconProps> = ({ name, ...props }) => {
  return <IconRoot {...props}>{renderIconImg(name)}</IconRoot>;
};
