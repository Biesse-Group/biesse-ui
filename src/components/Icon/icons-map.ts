import * as IconSources from "./icon-sources";

// Must be update each time a new icon is inserted in the library
const iconsMap = {
  arrow: IconSources.ArrowIcon,
  "arrow-right": IconSources.ArrowRightIcon,
  "arrow-left": IconSources.ArrowLeftIcon,
  careers: IconSources.CareersIcon,
  "material-wood": IconSources.CategoryWoodIcon,
  "material-metal": IconSources.CategoryMetalIcon,
  "material-composite": IconSources.CategoryCompositeIcon,
  "material-glass": IconSources.CategoryGlassIcon,
  "material-stone": IconSources.CategoryStoneIcon,
  "chevron-right": IconSources.ChevronRightIcon,
  contacts: IconSources.ContactsIcon,
  country: IconSources.CountryIcon,
  customer: IconSources.CustomerIcon,
  "fast-repair": IconSources.RepairFastIcon,
  hamburger: IconSources.HamburgerIcon,
  innovation: IconSources.InnovationIcon,
  location: IconSources.LocationIcon,
  "place-holder": IconSources.PlaceHolderIcon,
  search: IconSources.SearchIcon,
  sustainability: IconSources.SustainabilityIcon,
  "spare-parts": IconSources.SparePartsIcon,
};

export default iconsMap;
