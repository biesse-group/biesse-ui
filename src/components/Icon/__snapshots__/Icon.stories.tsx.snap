// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons/Icon ExtraSmall smoke-test 1`] = `
<span class="sc-aXZVg HgUIg">
  <svg viewbox="0 0 50 50"
       xmlns="http://www.w3.org/2000/svg"
       xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path d="M0 0h23.795v23.795H0z">
      </path>
    </defs>
    <g fill="none"
       fill-rule="evenodd"
    >
      <path d="M25.267 46.753c2.993 0 5.812.795 8.247 2.173l13.5-13.5c-4.447-3.9-9.886-6.644-15.872-7.786l-6.024 6.296-6.11-6.384C12.985 28.608 7.508 31.29 3 35.128L16.872 49a16.772 16.772 0 0 1 8.395-2.247Z"
            fill="currentColor"
      >
      </path>
      <g transform="translate(13.308 1)">
        <path d="M23.795 11.898c0 6.57-5.327 11.897-11.897 11.897C5.327 23.795 0 18.468 0 11.898 0 5.327 5.327 0 11.898 0c6.57 0 11.897 5.327 11.897 11.898"
              fill="currentColor"
        >
        </path>
      </g>
    </g>
  </svg>
</span>
`;

exports[`Icons/Icon Large smoke-test 1`] = `
<span class="sc-aXZVg ipqFja">
  <svg viewbox="0 0 50 50"
       xmlns="http://www.w3.org/2000/svg"
       xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path d="M0 0h23.795v23.795H0z">
      </path>
    </defs>
    <g fill="none"
       fill-rule="evenodd"
    >
      <path d="M25.267 46.753c2.993 0 5.812.795 8.247 2.173l13.5-13.5c-4.447-3.9-9.886-6.644-15.872-7.786l-6.024 6.296-6.11-6.384C12.985 28.608 7.508 31.29 3 35.128L16.872 49a16.772 16.772 0 0 1 8.395-2.247Z"
            fill="currentColor"
      >
      </path>
      <g transform="translate(13.308 1)">
        <path d="M23.795 11.898c0 6.57-5.327 11.897-11.897 11.897C5.327 23.795 0 18.468 0 11.898 0 5.327 5.327 0 11.898 0c6.57 0 11.897 5.327 11.897 11.898"
              fill="currentColor"
        >
        </path>
      </g>
    </g>
  </svg>
</span>
`;

exports[`Icons/Icon Medium smoke-test 1`] = `
<span class="sc-aXZVg bMcScH">
  <svg viewbox="0 0 50 50"
       xmlns="http://www.w3.org/2000/svg"
       xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path d="M0 0h23.795v23.795H0z">
      </path>
    </defs>
    <g fill="none"
       fill-rule="evenodd"
    >
      <path d="M25.267 46.753c2.993 0 5.812.795 8.247 2.173l13.5-13.5c-4.447-3.9-9.886-6.644-15.872-7.786l-6.024 6.296-6.11-6.384C12.985 28.608 7.508 31.29 3 35.128L16.872 49a16.772 16.772 0 0 1 8.395-2.247Z"
            fill="currentColor"
      >
      </path>
      <g transform="translate(13.308 1)">
        <path d="M23.795 11.898c0 6.57-5.327 11.897-11.897 11.897C5.327 23.795 0 18.468 0 11.898 0 5.327 5.327 0 11.898 0c6.57 0 11.897 5.327 11.897 11.898"
              fill="currentColor"
        >
        </path>
      </g>
    </g>
  </svg>
</span>
`;

exports[`Icons/Icon Small smoke-test 1`] = `
<span class="sc-aXZVg cgGILR">
  <svg viewbox="0 0 50 50"
       xmlns="http://www.w3.org/2000/svg"
       xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <path d="M0 0h23.795v23.795H0z">
      </path>
    </defs>
    <g fill="none"
       fill-rule="evenodd"
    >
      <path d="M25.267 46.753c2.993 0 5.812.795 8.247 2.173l13.5-13.5c-4.447-3.9-9.886-6.644-15.872-7.786l-6.024 6.296-6.11-6.384C12.985 28.608 7.508 31.29 3 35.128L16.872 49a16.772 16.772 0 0 1 8.395-2.247Z"
            fill="currentColor"
      >
      </path>
      <g transform="translate(13.308 1)">
        <path d="M23.795 11.898c0 6.57-5.327 11.897-11.897 11.897C5.327 23.795 0 18.468 0 11.898 0 5.327 5.327 0 11.898 0c6.57 0 11.897 5.327 11.897 11.898"
              fill="currentColor"
        >
        </path>
      </g>
    </g>
  </svg>
</span>
`;
